internal/target/dtrack_target.go

```diff
@@ -40,85 +39,40 @@ func (g *DependencyTrackTarget) ValidateConfig() error {
 }
 
 func (g *DependencyTrackTarget) Initialize() {
-	client, _ := dtrack.NewClient(g.baseUrl, dtrack.WithAPIKey(g.apiKey))
-	g.imageToProject = make(map[string]uuid.UUID)
-
-	const pageSize = 10
-	pageNumber := 1
-	for {
-		projectsPage, err := client.Project.GetAll(context.TODO(), dtrack.PageOptions{
-			PageNumber: pageNumber,
-			PageSize:   pageSize,
-		})
-		if err != nil {
-			logrus.Errorf("Could not fetch projects: %v", err)
-			return
-		}
-
-		for _, project := range projectsPage.Projects {
-			for _, property := range project.Properties {
-				if property.Name == "image-name" {
-					g.imageToProject[property.Value] = project.UUID
-				}
-			}
-		}
-
-		if pageNumber*pageSize >= projectsPage.TotalCount {
-			break
-		}
-
-		pageNumber++
-	}
 }
 
-func (g *DependencyTrackTarget) ProcessSbom(imageID, sbom string) {
-	if sbom == "" {
-		logrus.Infof("Empty SBOM - skip image (image=%s)", imageID)
+func (g *DependencyTrackTarget) ProcessSbom(image kubernetes.ContainerImage, sbom string) {
+	fullRef, err := parser.Parse(image.Image)
+	if err != nil {
+		logrus.WithError(err).Errorf("Could not parse imageID %s", image.ImageID)
 		return
 	}
 
-	client, _ := dtrack.NewClient(g.baseUrl, dtrack.WithAPIKey(g.apiKey))
-	logrus.Infof("Sending SBOM to Dependency Track (image=%s)", imageID)
+	imageName := fullRef.Repository()
+	tagName := fullRef.Tag()
+	if tagName == "" {
+		tagName = "latest"
+	}
```

Comment:
The fallback can be removed, the library has it built-in: https://github.com/novln/docker-parser/blob/master/parser_test.go#L25

Reply:
Changed the two things.